docs: adapt lifecycle-management and observability getting started guides to use KeptnAppContext (#2880)
keptn/lifecycle-toolkit: docs/docs/getting-started/lifecycle-management.md

> ```yaml

Reviewer: since we are touching this file I would suggest migrating yamls in separate files

Author: we have a special ticket for that https://github.com/keptn/lifecycle-toolkit/issues/2587 I would say I won't move the code blocks to separate files unless I directly touch them

Reviewer: but you just did 😢

Author: I adapted those that I touched

> Note: If a pre-deployment task fails, the `Pod` remains in a Pending state.

## More control over the application

To customize checks associated with the application, we can create a `KeptnAppContext` resource and define
a set of pre/post deployment tasks or evaluations for the whole application.
Note that the name of the `KeptnAppContext` resource needs to match the name of the automatically
created `KeptnApp` and the value present in the `keptn.sh/app` or `app.kubernetes.io/part-of`
annotations.
In this case it needs to be `keptndemoapp`.
An example of `KeptnAppContext`:

```yaml
apiVersion: lifecycle.keptn.sh/v1beta1
kind: KeptnAppContext
metadata:
  name: keptndemoapp
  namespace: keptndemo
spec:
  postDeploymentTasks:
  - send-event
```

> This way the `send-event` task will be executed after the deployment of the whole application

Reviewer 2: ```suggestion
This way, the `send-event` task is executed after the deployment of the whole application;
```

> (all of the workloads present in the `KeptnApp` are un a `Running` state).

Reviewer 2: ```suggestion
in other words, after all of the workloads present in the `KeptnApp` are in a `Running` state).
```

> A detailed description of all the available fields of `KeptnAppContext` resource can be find in the

Reviewer 2: ```suggestion
A detailed description of all the available fields of the `KeptnAppContext` resource can be found in the
```

> Note: If a pre-deployment task fails, the `Pod` remains in a Pending state.

## More control over the application

To customize checks associated with the application, we can create a `KeptnAppContext` resource and define
a set of pre/post deployment tasks or evaluations for the whole application.
Note that the name of the `KeptnAppContext` resource needs to match the name of the automatically
created `KeptnApp` and the value present in the `keptn.sh/app` or `app.kubernetes.io/part-of`
annotations.
In this case it needs to be `keptndemoapp`.
An example of `KeptnAppContext`:

> ```yaml

Reviewer: please use includes

Author: done
